packages/@react-aria/focus/src/useFocusRing.ts

```diff
@@ -34,19 +34,16 @@ interface FocusRingAria {
  * not with a mouse, touch, or other input methods.
  */
 export function useFocusRing(props: FocusRingProps = {}): FocusRingAria {
-  let {isTextInput, within} = props;
+  let {autoFocus, isTextInput, within} = props;
   let [isFocused, setFocused] = useState(false);
   let [isFocusWithin, setFocusWithin] = useState(false);
-  let [isFocusVisible, setFocusVisible] = useState(false);
+  let [isFocusVisibleState, setFocusVisible] = useState(autoFocus || isFocusVisible());
 
   // trigger on isFocusVisible state change when component is focused and modality changes
   // or when the component loses focus
-  useFocusVisibilityEmitter((isFocusVisible, modality, e) => {
-    if (!isKeyboardFocusEvent(isTextInput, modality, e)) {
-      return;
-    }
-    setFocusVisible((within ? isFocusWithin : isFocused) && isFocusVisible);
-  }, [within, isFocusWithin, isFocused]);
+  useFocusVisibleListener((isFocusVisible) => {
+    setFocusVisible(isFocusVisible);
```

Comment:
Why did this change? Won't this now have the same problem as the original implementation? It will update the state for _all_ focus rings whenever the focus visibility state changes, rather than only the actually focused element.

Reply:
Ah, need it both here and in the return object